compiler/src/main/scala/org/scalajs/core/compiler/PrepJSInterop.scala

```diff
@@ -1060,13 +1087,25 @@ abstract class PrepJSInterop extends plugins.PluginComponent
   /** Checks that argument to @JSName on [[sym]] is a literal.
    *  Reports an error on each annotation where this is not the case.
    */
-  private def checkJSNameLiteral(sym: Symbol): Unit = {
+  private def checkJSNameArgument(sym: Symbol): Unit = {
     for {
       annot <- sym.getAnnotation(JSNameAnnotation)
-      if annot.stringArg(0).isEmpty
+      argTree <- annot.args.headOption
```

Comment:
Any reason not to use `annot.args.head` here? If it doesn't have any argument, it should have failed to typecheck prior to getting to this phase.